cmd/fuseml_core/wire.go

```diff
@@ -0,0 +1,72 @@
+// +build wireinject
+
+package main
+
+import (
+	"log"
+
+	"github.com/google/wire"
+
+	"github.com/fuseml/fuseml-core/gen/application"
+	"github.com/fuseml/fuseml-core/gen/codeset"
+	"github.com/fuseml/fuseml-core/gen/project"
+	"github.com/fuseml/fuseml-core/gen/runnable"
+	"github.com/fuseml/fuseml-core/gen/version"
+	"github.com/fuseml/fuseml-core/gen/workflow"
+	"github.com/fuseml/fuseml-core/pkg/core"
+	"github.com/fuseml/fuseml-core/pkg/core/gitea"
+	"github.com/fuseml/fuseml-core/pkg/core/manager"
+	"github.com/fuseml/fuseml-core/pkg/core/tekton"
+	"github.com/fuseml/fuseml-core/pkg/domain"
+	"github.com/fuseml/fuseml-core/pkg/svc"
+)
+
+var storeSet = wire.NewSet(
+	core.NewApplicationStore,
+	wire.Bind(new(domain.ApplicationStore), new(*core.ApplicationStore)),
+	gitea.NewAdminClient,
+	wire.Bind(new(domain.GitAdminClient), new(*gitea.AdminClient)),
+	core.NewGitCodesetStore,
+	wire.Bind(new(domain.CodesetStore), new(*core.GitCodesetStore)),
+	core.NewGitProjectStore,
+	wire.Bind(new(domain.ProjectStore), new(*core.GitProjectStore)),
+	core.NewRunnableStore,
+	wire.Bind(new(domain.RunnableStore), new(*core.RunnableStore)),
+	core.NewWorkflowStore,
+	wire.Bind(new(domain.WorkflowStore), new(*core.WorkflowStore)),
```

Comment:
Is it just me, or does wire also force you to use golang best practices regarding the use of interfaces ?